coalib/results/SourcePosition.py

```diff
@@ -36,3 +37,8 @@ def __json__(self, use_relpath=False):
         if use_relpath:
             _dict['file'] = relpath(_dict['file'])
         return _dict
+
+    def __str__(self):
+        return ('file=' + self.filename +
```

Comment:
getting very close ... so now ... wtf is this string.

why did you chose this string instead of the one I suggested?

do not just add what I suggested instead.

Please defend your choice.  Explain it.  Maybe as a code comment, or here as a comment in GitHub.

Once this is merged, this is the string representation of a SourcePosition object --- any change to it after it is merged is a BIG BAD BREAKING CHANGE.

So we need to be sure this is the best , most stable , most appropriate, way to convert a SourcePosition into a string.

This is [serious business](https://www.urbandictionary.com/define.php?term=serious%20business)! :P

Reply:
What you suggested was to separate it by colons but it think it is more natural to separate it by semi-colon. 
Also about only printing line and column if it exists, even if it doesn't exist printing it will do no harm.
I will remove `file=` but `line=` and `column=` should be kept to know which is what.

Reply:
Btw, why not comma :P

Reply:
Even comma is fine.

Reply:
`filename:lineno:colno` is dense, but is commonly understood

and it has no i18n issues.  words in a `__str__` are bad news for i18n, as a str(..) must be the same for every user, so they cant be internationalised, and they force English onto all users, or force parsers creating by programmers to use English keywords to parse the output of `str(..)` before adding i18n in order to show something localised for a non-English UI.